### PR TITLE
Handle text formats in style manager

### PR DESCRIPTION
**Pull request by nyalldawson**
This allows the style manager to be used to manage text formats, which store the font, color, buffers, shadows, and backgrounds of text formats (i.e. all the formatting parts of label settings, which can be used outside of labels, eg in layouts).

Text Formats are added alongside the existing symbols and color ramp management options, and offer identical functionality as these.

A new text format list widget has been added below the first tab of label settings, which shows a filterable list of text formats stored within the user's style manager database. It's the equivalent of the symbol 
library which is shown below symbol settings when styling vector layers.

This allows users to easily set label format to match styles saved in the local style database, and also to add a new text format to the style database based on the current label formatting settings.

(In order to make room for this, some lesser-used label settings were moved from the first tab to the "formatting" tab, under @nirvn's guidance)

Thanks to SLYR, and closing the ever-decreasing feature gap between QGIS and ArcGIS' vector symbology featureset...

Screenies:

image

image

### CODE REVIEW

**Comment by nyalldawson** [comment]
Bonus feature -- now comes with added support for Label Settings in style manager! I've been using this all day doing some heavy cartography work and it makes A HUGEEEEEEEEEEEEEEEE usability difference. :tada:

**Comment by nirvn** [comment]
I can attest, I've been testing this branch for a day or so, it's a game changing experience.

File: python/gui/auto_generated/symbology/qgsstylesavedialog.sip.in [hunk 1]
```diff
@@ -33,20 +33,32 @@ Constructor for :py:class:`QgsSymbolSaveDialog`
...
+%Docstring
+Sets the default ``tags`` for the newly created item.
+
+.. versionadded:: 3.10
 %End
 
     QString tags() const;
 %Docstring
-returns the text value of the tags element
+Returns any tags entered for the new symbol (as a comma seperated value list).
```
Comment (nyalldawson) [inline python/gui/auto_generated/symbology/qgsstylesavedialog.sip.in]:
```suggestion
Returns any tags entered for the new symbol (as a comma separated value list).
```

File: src/gui/symbology/qgsstylesavedialog.h [hunk 2]
```diff
@@ -43,16 +43,37 @@ class GUI_EXPORT QgsStyleSaveDialog: public QDialog, private Ui::QgsStyleSaveDia
...
-    //! returns the text value of the tags element
+    /**
+     * Sets the default \a tags for the newly created item.
+     *
+     * \since QGIS 3.10
+     */
+    void setDefaultTags( const QString &tags );
+
+    /**
+     * Returns any tags entered for the new symbol (as a comma seperated value list).
```
Comment (nyalldawson) [inline src/gui/symbology/qgsstylesavedialog.h]:
```suggestion
     * Returns any tags entered for the new symbol (as a comma separated value list).
```

File: tests/src/python/test_qgsstylemodel.py [hunk 3]
```diff
@@ -789,7 +1546,12 @@ def testSetData(self):
         self.assertTrue(model.setData(model.index(1, 0), 'ramp new name', Qt.EditRole))
         self.assertEqual(model.data(model.index(1, 0), Qt.DisplayRole), 'ramp new name')
         self.assertEqual(style.colorRampNames(), ['ramp new name'])
-
+        self.assertTrue(model.setData(model.index(2, 0), 'format new name', Qt.EditRole))
+        self.assertEqual(model.data(model.index(2, 0), Qt.DisplayRole), 'format new name')
+        self.assertEqual(style.textFormatNames(), ['format new name'])
+        self.assertTrue(model.setData(model.index(3, 0), 'settings new name', Qt.EditRole))
+        self.assertEqual(model.data(model.index(3, 0), Qt.DisplayRole), 'settings new name')
+        self.assertEqual(style.labelSettingsNames(), ['settings new name'])
```
Comment (nyalldawson) [inline tests/src/python/test_qgsstylemodel.py]:
```suggestion
        self.assertEqual(style.labelSettingsNames(), ['settings new name'])

```

**Pull request merged**